app/src/main/java/space/taran/arknavigator/mvp/view/item/FileItemViewHolder.kt

```diff
@@ -19,11 +19,66 @@ class FileItemViewHolder(private val binding: ItemFileGridBinding) :
 
     override fun setIcon(icon: Preview): Unit = with(binding.root) {
         Log.d(ITEMS_CONTAINER, "setting icon $icon")
-        if (icon.predefined != null) {
-            binding.iv.setImageResource(imageForPredefinedIcon(icon.predefined))
-            binding.iv.imageTintList = ColorStateList.valueOf(ContextCompat.getColor(context, R.color.gray))
+        if (icon.predefinedIcon != null) {
+            binding.iv.setImageResource(imageForPredefinedIcon(icon.predefinedIcon))
         } else {
-            loadImage(icon.image!!, binding.iv)
+            when (icon.fileType) {
+                FileType.GIF -> loadGifThumbnailWithPlaceHolder(
+                    icon.previewPath!!,
+                    imageForPredefinedExtension("gif"),
+                    binding.iv)
+                FileType.PDF -> {
+                    //Temporary workaround for asynchronous loading.
```

Comment:
What makes this code temporary? If we load the preview in lazy manner, then we should display progress spinner.